citar.el

```diff
@@ -179,6 +193,25 @@ All functions that match a particular field are run in order."
   :type '(alist :key-type   (choice (const t) (repeat string))
                 :value-type function))
 
+(defcustom citar-prefilter-entries '(nil . t)
```

Comment:
@roshanshariff WDYT of this idea of mine?

This would address #449. Earlier I had closed it because I couldn't figure out a flexible way to do this, but this seems a good approach?

Reply:
I think it's better to implement a Consult-like narrowing system. I looked at its code, and behind the scenes it maintains a list of predicates that are combined into `minibuffer-completion-predicate`. We'd bind some keys to allow predicates to be enabled and disabled on-the-fly, just like Consult does.

Then we could make `citar-select-ref` activate a certain predicate by default, but you'd always be able to press a key to remove the filter and/or add other filters.

Reply:
I agree that would be valuable, without us requiring consult.

But this (really the car) is orthogonal, and is just saying whether to pre-filter with `citar-open-notes` and such, as per #449.

In any case, feel free to remove or adapt this at will.